src/common/Configuration/Config.h

```diff
@@ -17,40 +19,46 @@ class ConfigMgr
     ~ConfigMgr() = default;
 
 public:
-    static ConfigMgr* instance();
-
-    /// Method used only for loading main configuration files (authserver.conf and worldserver.conf)
-    bool LoadInitial(std::string const& file);
+    bool LoadAppConfigs();
+    bool LoadModulesConfigs();
+    void Configure(std::string const& initFileName, std::vector<std::string> args, std::string const& modulesConfigList = "");
 
-    /**
-     * This method loads additional configuration files
-     * It is recommended to use this method in WorldScript::OnConfigLoad hooks
-     *
-     * @return true if loading was successful
-     */
-    bool LoadMore(std::string const& file);
+    static ConfigMgr* instance();
 
     bool Reload();
 
-    bool LoadAppConfigs(std::string const& applicationName = "worldserver");
-    bool LoadModulesConfigs();
+    std::string const& GetFilename();
+    std::string const GetConfigPath();
+    std::vector<std::string> const& GetArguments() const;
+    std::vector<std::string> GetKeysByString(std::string const& name);
 
-    std::string GetStringDefault(std::string const& name, const std::string& def, bool logUnused = true);
-    bool GetBoolDefault(std::string const& name, bool def, bool logUnused = true);
-    int GetIntDefault(std::string const& name, int def, bool logUnused = true);
-    float GetFloatDefault(std::string const& name, float def, bool logUnused = true);
+    template<class T>
+    T GetOption(std::string const& name, T const& def, bool showLogs = true) const;
```

Comment:
@Winfidonarleyan I like the approach using generic types (templates). However, I think we should also keep the old methods and mark them as deprecated so we won't break modules and can give the opportunity to migrate them without failures. EXAMPLE:

```cpp
// @deprecated DO NOT USE - use GetOption<float> instead.
float GetFloatDefault(std::string const& name, float def, bool logUnused = true)
{
    return GetOption<float>(name, def, logUnused);
}
```

Reply:
Or the owner of the modules update to the new standard.

Perhaps it could be kept for a transitional period, a couple of weeks. But after that imo we dont need duplicate code

Reply:
yes it's definitely for a transitional period, but 2 weeks is not enough